lib/graphql/type/schema.ex

```diff
@@ -81,6 +82,19 @@ defmodule GraphQL.Schema do
     end
   end
 
+  defp reduce_types(typemap, %Input{} = type) do
+    if Map.has_key?(typemap, type.name) do
+      typemap
+    else
+      typemap = Map.put(typemap, type.name, type)
+      thunk_fields = CompositeType.get_fields(type)
+      typemap = Enum.reduce(thunk_fields, typemap, fn({_,fieldtype},typemap) ->
```

Comment:
We should, however I'm about to start cleaning up some of the 1.3 warnings so will merge this as is and deal with it in a separate commit.